lang/py/src/avro/schema.py

```diff
@@ -405,6 +424,83 @@ def __eq__(self, that):
     return self.props == that.props
 
 #
+# Decimal Type
+#
+class DecimalSchema(LogicalSchema):
+  def __init__(self, decimal_type, precision, scale=0, size=0, fixed_schema=None):
+    if decimal_type != 'fixed' and decimal_type != 'bytes':
```

Comment:
Is it possible to split these implementation (fixed and bytes) into separate classes or at least two separate constructors? It'd make the interface much clear and the validation is easier. And why does it take a `fixed_schema` as a parameter?